CONTRIBUTING.md

```diff
@@ -0,0 +1,189 @@
+```
+
+With pip/venv, you can create a virtual environment with
+
+```bash
+python -m venv qtpy-env
+```
+
+And activate it with the following on Linux and macOS,
+
+```bash
+source qtpy-env/bin/activate
+```
+
+or on Windows (cmd),
+
+```cmd
+.\qtpy-env\Scripts\activate.bat
+```
+
+Regardless of the tool you use, make sure to remember to always activate your environment before using it.
+
+
+### Install a Python Qt binding
+
+Before installing QtPy itself, make sure you have the Qt binding(s) you wish to develop against.
+For example, for PyQt5 on Conda, you'd run:
+
+```bash
+conda install -c conda-forge pyqt=5
+```
+
+Or for the same using pip, you'd execute:
+
+```bash
+python -m pip install pyqt5==5.* PyQtWebEngine==5.*
+```
+
+While having separate environments for each binding is recommended, you can install multiple in one environment and select between them using the ``QT_API`` environment variable, as described in the README (for example, setting it to ``pyqt5`` to select PyQt5, if it is installed).
+
+
+### Install QtPy in editable mode
+
+Finally, to install the QtPy package itself in editable ("development") mode, where updates to the source files will be reflected in the installed package, and include any additional dependencies used for development, run
+
+```bash
+python -m pip install -e .[test]
+```
+
+You can then import and use QtPy as normal.
+When you make changes in your local copy of the git repository, they will be reflected in your installed copy as soon as you re-run Python.
+
+
+
+## Deciding Which Branch to Use
+
+When you start to work on a new pull request (PR), you need to be sure that your work is done on top of the correct branch, and that you base your PR on Github against it.
+
+To guide you, issues on Github are marked with a milestone that indicates the correct branch to use.
+If not, follow these guidelines:
+
+* Use the latest release branch (e.g. ``1.x``) to fix security issues and critical bugs only (if in any doubt, ask first)
+* Use ``master`` branch for anything else, particularly introducing new features or breaking compatibility with previous versions
+
+Of course, if a bug is only present in ``master``, please base bugfixes on that branch.
+
+
+
+## Making Your Changes
+
+To start working on a new PR, you need to execute these commands, filling in the branch names where appropriate (``<BASE-BRANCH>`` is the branch you're basing your work against, e.g. ``master``, while ``<FEATURE-BRANCH>`` is the branch you'll be creating to store your changes, e.g. ``fix-startup-bug`` or ``add-widget-support``:
+
+```bash
+git checkout <BASE-BRANCH>
+git pull upstream <BASE-BRANCH>
+git checkout -b <FEATURE-BRANCH>
+```
+
+Once you've made and tested your changes, commit them with a descriptive message of 74 characters or less written in the imperative tense, with a capitalized first letter and no period at the end.
+For example:
+
+```bash
+git commit -am "Fix bug importing package on Windows"
+```
```

Comment:
Should we expand here more about the expected conditions on the commit history to accept a PR @CAM-Gerlach ?

Reply:
Thanks, I added some more tips and suggestions for commit messages and structure, along with a few other things, while also making clear that these are more guidelines than strict rules so not to worry to much about it, and if there is a serious problem we're happy to help (e.g. with a squash or rebase, though pending the release of the reviewer guidelines and your feedback on them I didn't want to set a hard policy there yet).